std-bits/table/src/main/java/org/enso/table/read/QuoteHelper.java

```diff
@@ -0,0 +1,54 @@
+package org.enso.table.read;
+
+import java.util.function.Consumer;
+
+/**
+ * A helper class providing functionality of parsing optionally quoted strings and detecting
+ * mismatched quotes.
+ *
+ * <p>The tool is meant for use with the output of the {@code CsvParser} with the setting {@code
+ * keepQuotes} set to true. The strings which were quoted will keep the enclosing quotes, but any
+ * escaped quotes within the string are correctly parsed (for example, if the escaped quote is `\"`,
+ * it will appear as already parsed `"` quote inside of the string, so no escape processing is
+ * performed at this stage, because it was done earlier by the parser.
+ */
+public class QuoteHelper {
+  private final Consumer<String> mismatchedQuoteCallback;
+  private final char quoteCharacter;
+
+  public QuoteHelper(Consumer<String> mismatchedQuoteCallback, char quoteCharacter) {
+    this.mismatchedQuoteCallback = mismatchedQuoteCallback;
+    this.quoteCharacter = quoteCharacter;
+  }
+
+  /**
+   * Strips enclosing quotes from the string, if they are present.
+   *
+   * <p>If the quote appears only at one end of the string, a mismatched quote callback is called to
+   * report the issue.
+   */
+  public String stripQuotes(String text) {
```

Comment:
I would make this a static method. 
```java
public string String stripQuotes(String text, char quoteCharacter, Consumer<String> mismatchedQuoteCallback)
```

Currently we are creating and freeing a new object for every row of the dataset we read which seem unnecessary.